nupic/encoders/scalar.capnp

```diff
@@ -0,0 +1,20 @@
+@0xfa7d16f86048a6e4;
+
+struct ScalarEncoderProto {
+  w @0 :Int32;
+  minval @1 :Float32;
+  maxval @2 :Float32;
+  periodic @3 :Bool;
+  n @4 :Int32;
+  radius @5 :Float32;
+  resolution @6 :Float32;
+  name @7 :Text;
+  verbosity @8 :Int8;
+  clipInput @9 :Bool;
+
+  halfwidth @10 :Int32;
```

Comment:
Ultimately, I think that `halfwidth` and the others that are normally derived in `__init__()` and `_initEncoder()` should be deprecated in the schema, but in this implementation, we're bypassing the `__init__()` method, so I opted to do a straight mapping of values for the full state of the object.  The current `__init__()` and `_initEncoder()` methods need to know the original arguments to the constructor, which aren't preserved in the serialization, and fixing that up would expand the scope of this task.

I'd rather do as you suggest and remove `halfwdith`, etc., but only after `__init__()` and `_initEncoder()` are fixed as I describe.